File: mobile/ios/ShiftPlanner/ShiftPlanner/Features/Employee/Availability/Repositories/AvailabilityRepository.swift
import Foundation

protocol AvailabilityRepository {
    func fetchAvailability(employeeId: Int) async throws -> EmployeeAvailabilityResponseDTO
    func saveAvailability(employeeId: Int, payload: EmployeeAvailabilityUpsertDTO) async throws -> EmployeeAvailabilityResponseDTO
}

File: mobile/android/app/src/main/java/com/froggyriia/shiftplanner/data/availability/AvailabilityRepository.kt
package com.froggyriia.shiftplanner.data.availability

import com.froggyriia.shiftplanner.domain.model.EmployeeAvailabilityData

interface AvailabilityRepository {
    suspend fun fetchAvailability(employeeId: Int): EmployeeAvailabilityData
    suspend fun saveAvailability(
        employeeId: Int,
        data: EmployeeAvailabilityData
    ): EmployeeAvailabilityData
}
